### ISSUE TITLE
Interact with Mamba on Electron

### ISSUE THREAD

**Issue by itsjoeoui** [mamba-org/mamba#3079]
The following is a quick summary of my conversation with @jjerphan 

The context: we are developing an Electron desktop app at @flojoy-ai that requires a local Python environment to function.

We started trying to use conda/mamba/micromamba for environment management, but at the end resorted to Poetry instead, for the following reason:

"When you call conda/manba, it actually invokes the shell function. We are using ChildProcess.exec to execute shell commands in Electron. This means that every time we want to execute something we need to define that conda/manba function again, which is quite not convenient and error-prone in my opinion."

Where as with Poetry, we simply needs to install it with Pipx, then ensure it is in user's path, and then we are good to go, which is a much more hassle free setup compared to conda/mamba.

It is very likely that there is a better way to do this and I did not discover it, would be much appreciated if someone can shed more light into this issue 🙂

**Comment by jonashaag** [comment]
You can use `conda/mamba/micromamba run`. That doesn't require shell activation.

Alternatively you can use

```
bash -c 'eval "$(micromamba shell hook)"; micromamba activate <env>; ...'
```

but that's almost exactly the same as using `run`

**Comment by itsjoeoui** [comment]
Hey @jonashaag !

Appreciate the quick reply! I remember I have tried `micromamba run` when I was still working on this a couple months ago, but then I encountered some other issues instead, which I can't remember right on top of my head right now unfortunately.

I will follow up on this issue here if we ever revisit mamba/micromamba, but most likely that won't be any time soon since we have already got a pretty stable Poetry setup.
